tests/AerospikeCacheTest.php

```diff
@@ -3,7 +3,200 @@
 
 namespace Lmc\AerospikeCache;
 
-class AerospikeCacheTest
+use PHPUnit\Framework\TestCase;
+
+class AerospikeCacheTest extends TestCase
 {
+    /**
+     * @dataProvider provideStatusCodes
+     */
+    public function testShouldConfirmItemPresence(int $statusCode, bool $expectedReturnedValue): void
+    {
+        $aerospikeMock = $this->createMock(\Aerospike::class);
+        $aerospikeCache = new AerospikeCache($aerospikeMock, 'test', 'cache');
+
+        $aerospikeKey = ['ns' => 'test', 'set' => 'cache', 'key' => 'foo'];
+        $aerospikeMock->expects($this->once())
+            ->method('initKey')
+            ->willReturn($aerospikeKey);
+
+        $aerospikeMock->expects($this->once())
+            ->method('get')
+            ->with($aerospikeKey)
+            ->willReturn($statusCode);
+
+        $hasItem = $aerospikeCache->hasItem('foo');
+
+        $this->assertSame($expectedReturnedValue, $hasItem);
+    }
+
+    /**
+     * @dataProvider provideStatusCodes
+     */
+    public function testShouldSaveItem(int $aerospikeStatusCode, bool $expectedSaveSuccessful): void
```

Comment:
I'd like to see a test to cover failed keys as well.. could you add it?

Reply:
The provider provideStatusCodes always has 2 cases. Success and fail.

Reply:
But it doesn't test failed keys - which I'd like to see checked.

Reply:
That would be a functional scenario that is testing the behaviour of getting the cacheItem (which coul be hit or not) and saving it.
The PSR16 use of cache is 
`//create the cache Item. You dont create it yourself but with getItem method
$cacheItem = $cache->getItem();
$cache->save($cacheItem);`

Mocking a result from a "get" method was the easiest way to get correct CacheItem object.

Reply:
I've send PR for this directly to @czhDavid fork - https://github.com/czhDavid/aerospike-cache-php/pull/1